src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include <dwrite_3.h>
+#include <AppInstallerStrings.h>
+#include <winget/Fonts.h>
+#include <winget/Locale.h>
+
+namespace AppInstaller::Fonts
+{
+    namespace
+    {
+        std::vector<std::filesystem::path> GetFontFilePaths(const wil::com_ptr<IDWriteFontFace>& fontFace)
+        {
+            UINT32 fileCount = 0;
+            THROW_IF_FAILED(fontFace->GetFiles(&fileCount, nullptr));
+
+            static_assert(sizeof(wil::com_ptr<IDWriteFontFile>) == sizeof(IDWriteFontFile*));
+            std::vector<wil::com_ptr<IDWriteFontFile>> fontFiles;
+            fontFiles.resize(fileCount);
+
+            THROW_IF_FAILED(fontFace->GetFiles(&fileCount, fontFiles[0].addressof()));
+
+            std::vector<std::filesystem::path> filePaths;
+            for (UINT32 i = 0; i < fileCount; ++i) {
+                wil::com_ptr<IDWriteFontFileLoader> loader;
+                THROW_IF_FAILED(fontFiles[i]->GetLoader(loader.addressof()));
+
+                const void* fontFileReferenceKey;
+                UINT32 fontFileReferenceKeySize;
+                THROW_IF_FAILED(fontFiles[i]->GetReferenceKey(&fontFileReferenceKey, &fontFileReferenceKeySize));
+
+                if (const auto localLoader = loader.try_query<IDWriteLocalFontFileLoader>()) {
+                    UINT32 pathLength;
+                    THROW_IF_FAILED(localLoader->GetFilePathLengthFromKey(fontFileReferenceKey, fontFileReferenceKeySize, &pathLength));
+                    pathLength += 1; // Account for the trailing null terminator during allocation.
+
+                    wchar_t* path = new wchar_t[pathLength];
+                    THROW_IF_FAILED(localLoader->GetFilePathFromKey(fontFileReferenceKey, fontFileReferenceKeySize, &path[0], pathLength));
+                    filePaths.push_back(std::move(std::wstring(path)));
+                }
+            }
+
+            return filePaths;
+        }
+
+        std::wstring GetLocalizedStringFromFont(const wil::com_ptr<IDWriteLocalizedStrings>& localizedStringCollection)
+        {
+            std::vector<std::string> locales = AppInstaller::Locale::GetUserPreferredLanguages();
+            std::wstring preferredLocale = Utility::ConvertToUTF16(!locales.empty() ? locales[0] : "en-US");
```

Comment:
My previous comment still stands; I don't think that defaulting to looking up `en-US` is correct.  If we can't find the preferred locale, we should just iterate through the list we have here (that seems easy enough to not leave the `//TODO`).

If we don't find one of the preferred locales, we should not use `en-US`, but rather just use index 0.

Additionally, from a performance perspective it is probably worth:
1. Creating a `std::wstring` version of `GetUserPreferredLanguages`
2. Holding onto the list of locales so that we don't have to create it so many times